ci/ctest_to_gitlab.sh

```diff
@@ -95,7 +95,7 @@ JOB_TEMPLATE="
   variables:
     SLURM_CPUS_PER_TASK: {{CPUS_PER_TASK}}
     SLURM_NTASKS: {{NTASKS}}
-    SLURM_TIMELIMIT: '20:00'
+    SLURM_TIMELIMIT: '55:00'
```

Comment:
So much longer?

Reply:
We can probably reduce it to 50, but unfortunately yes, tests are quite a bit longer (`RANK_6` for `cuda gcc11 codecov test` is 45 minutes). The reason is that the eigensolver tests are now 4x: I test the original API and the new API is tested with 0, half, and the full eigenspectrum. Maybe we can reduce the coverage, especially for the C API (only testing the old API and the new one with half of the eigenvectors)?

Reply:
@rasolca suggested that the C API tests could be separated out (like the miniapps in #1112).